src/qt/transactionrecord.cpp

```diff
@@ -61,14 +68,14 @@ QList<TransactionRecord> TransactionRecord::decomposeTransaction(const CWallet *
 
             TransactionRecord sub(hash, nTime);
             CTxDestination address;
-
+            
             if (mine) {
                 sub.involvesWatchAddress = mine & ISMINE_WATCH_ONLY;
                 if (ExtractDestination(txout.scriptPubKey, address) && IsMine(*wallet, address))
                 {
                     sub.type = TransactionRecord::SpendToSelf;
                     sub.address = CBitcoinAddress(address).ToString();
-                    sub.credit = txout.nValue;
+                    sub.credit = -(nTxFee / nOuts);
```

Comment:
This will produce an incorrect value if we spend to our address and other address in the same transaction. My idea is we should follow normal transaction behavior. What it does for normal transaction is it will append fee to the first recipient. Take a look here for it logic https://github.com/zcoinofficial/zcoin/blob/master/src/qt/transactionrecord.cpp#L168

My idea for how to handle spend to yourself is:

- If all recipients are our then show only one entry with debit by fee.
- If there is at least one other recipient then we will add fee to the debit of that entry. For our addresses in this case just show as credit.

Reply:
OK! Let's follow to your suggestions.